desertbot/modules/commands/Comic.py

```diff
@@ -0,0 +1,235 @@
+"""
+Created on Apr 18, 2019
+
+@author: lunik1
+"""
+import base64
+import glob
+import json
+from io import BytesIO
+from random import shuffle
+
+import requests
+from PIL import Image, ImageDraw, ImageFont
+from twisted.plugin import IPlugin
+from zope.interface import implementer
+
+from desertbot.message import IRCMessage
+from desertbot.moduleinterface import IModule
+from desertbot.modules.commandinterface import BotCommand
+from desertbot.response import IRCResponse, ResponseType
+
+
+@implementer(IPlugin, IModule)
+class Comic(BotCommand):
+    # just hardcode a limit on max messages for now, should limit comic size pretty effecively?
+    messageLimit = 8
+
+    def actions(self):
+        # define additional actions so all PRIVMSGs and ACTIONs in the channel get handled by storeMessage
+        return super(Comic, self).actions() + [("message-channel", 1, self.storeMessage),
+                                               ("action-channel", 1, self.storeMessage)]
+
+    def onLoad(self) -> None:
+        # messages need to be stored for comics to work, store in a list as (nick, message)
+        self.messageStore = []
+
+    def triggers(self):
+        return ['comic']
+
+    def help(self, query):
+        return 'comic - make a comic'
+
+    def execute(self, message: IRCMessage):
+        if len(self.messageStore) != 0:
+            comic = self.make_comic(self.messageStore)
+            return IRCResponse(ResponseType.Say, self.post_comic(comic), message.replyTo)
+
+    def storeMessage(self, message: IRCMessage):
+        """
+        Store the message into the messageStore, and prune it to the limit if needed
+        """
```

Comment:
this should filter out messages with .command set, and maybe any message starting with punctuation to catch other bot commands.
multiple punctuation should be kept, though, so ... can get into comics
and maybe allow quote marks?

it should also obey the ignore list, self.checkIgnoreList(message) returns True if the user who sent the message is ignored.